File: app/src/main/java/eu/darken/bluemusic/main/core/service/modules/RingVolumeModule.java
package eu.darken.bluemusic.main.core.service.modules;

import android.app.NotificationManager;

import javax.inject.Inject;

import eu.darken.bluemusic.main.core.audio.AudioStream;
import eu.darken.bluemusic.main.core.audio.StreamHelper;
import eu.darken.bluemusic.main.core.service.BlueMusicServiceComponent;
import eu.darken.bluemusic.settings.core.Settings;
import eu.darken.bluemusic.util.ApiHelper;

@BlueMusicServiceComponent.Scope
public class RingVolumeModule extends BaseVolumeModule {

    private final NotificationManager notMan;

    @Inject
    public RingVolumeModule(Settings settings, StreamHelper streamHelper, NotificationManager notMan) {
        super(settings, streamHelper);
        this.notMan = notMan;
    }

    @Override
    AudioStream.Type getType() {
        return AudioStream.Type.RINGTONE;
    }

    @Override
    public boolean areRequirementsMet() {
        return !ApiHelper.hasMarshmallow() || notMan.isNotificationPolicyAccessGranted();
    }
}

File: app/src/main/java/eu/darken/bluemusic/main/core/service/modules/events/RingVolumeModule.kt
package eu.darken.bluemusic.main.core.service.modules.events

import android.app.NotificationManager
import eu.darken.bluemusic.main.core.audio.AudioStream
import eu.darken.bluemusic.main.core.audio.StreamHelper
import eu.darken.bluemusic.main.core.service.BlueMusicServiceComponent
import eu.darken.bluemusic.settings.core.Settings
import eu.darken.bluemusic.util.ApiHelper
import javax.inject.Inject

@BlueMusicServiceComponent.Scope
class RingVolumeModule @Inject constructor(
        settings: Settings,
        streamHelper: StreamHelper,
        private val notMan: NotificationManager
) : BaseVolumeModule(settings, streamHelper) {

    override val type: AudioStream.Type = AudioStream.Type.RINGTONE

    override fun areRequirementsMet(): Boolean = !ApiHelper.hasMarshmallow() || notMan.isNotificationPolicyAccessGranted
}